string modeStr
float32 value
int32 waitTime
